test/cmd_line/read.test.ts

```diff
@@ -0,0 +1,24 @@
+"use strict";
+
+import { ModeHandler } from '../../src/mode/modeHandler';
+import { setupWorkspace, cleanUpWorkspace, assertEqualLines } from './../testUtils';
+import { runCmdLine } from '../../src/cmd_line/main';
+
+suite("read", () => {
+  let modeHandler: ModeHandler;
+
+  setup(async () => {
+    await setupWorkspace();
+    modeHandler = new ModeHandler();
+  });
+
+  teardown(cleanUpWorkspace);
+
+  test("Can read shell command output", async () => {
+    await runCmdLine('r! echo heyho', modeHandler);
```

Comment:
1. Can you replace the existing test with `:r! echo hey\\n\\tho`? Helps test more cases :)
2. Yeah, that's a backlog work item to display it on the status bar